woocommerce/google-listings-and-ads
# Require URL to match when syncing data to Merchant Center

### Changes proposed in this Pull Request:

This PR changes the product sync behaviour and requires the URL to match to the one we used to claim the website with. This ensures that the products are only able to sync on the original site, any staging site or copy of the site will not sync product data.

If the user wishes to change the site URL then the onboarding process must be completed again to perform a new site verification/claim.

Closes #1399

### Detailed test instructions:
1. Start with a live site that's fully onboarded and synced all products to the Merchant Center
2. Create a staging site (can use a plugin like [WP Staging](https://wordpress.org/plugins/wp-staging/))
3. Update a product on the staging site
4. Check the action scheduler logs and confirm that the product was not synced
5. Check the same product in the Merchant Center (might need to wait a while before it's updated) and confirm that the URL's are still from the original site
6. Use the filter `add_filter( 'woocommerce_gla_ready_for_syncing', '__return_true' );`
7. Update a product on the staging site
8. Check the action scheduler logs and confirm that this time the product did sync
9. Check the original site and confirm that product syncing continues to function as it was when updating products

### Additional details:

In addition to the tests for the added / changed code. There was no unit test present for the MerchantCenterService class, so I added it in this PR before adding the additional tests for this class.

### Changelog entry
* Fix - Prevent product sync if the site URL does not match the originally claimed URL.
